src/AutoRest.CSharp.V3/Generation/Writers/ClientWriter.cs

```diff
@@ -117,11 +117,11 @@ private void WriteManagementClientCtors(CodeWriter writer, Client client, Config
 
                     if (ManagementClientWriterHelpers.IsApiVersionParameter(parameter))
                     {
-                        writer.Append($"options.Version, ");
```

Comment:
Hmmm. You stumbled upon an awesome idea for the generator. Parameter ordering or optional parameters on calls wouldn't matter if we used named parameters everywhere. This should make calling writer logic easier and more consistent. However, the generated code might look more 'noisy'. Thoughts?

Reply:
I'm not a fan of the noisiness part. I would use it sparingly for cases where parameter order is not trivial to match. In this case I used it because I want to use the default value for `host` parameter but have to pass arguments that go both before and after it.

Reply:
What about... for any parameters that are defaulted, we use a named parameter. Then, it would be consistent and be very resistant to unintended bugs. We could have a parameter callsite writing (such as `:P`) that would do this automatically.

Reply:
I don't see the benefit yet, this change actually fixed a bug but doing it everywhere wouldn't.